### PR TITLE
Lps 74772

### PR DESCRIPTION
**Pull request by lanceji92674957**

### CODE REVIEW

File: modules/apps/collaboration/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/exportimport/data/handler/BlogsEntryStagedModelDataHandler.java [hunk 1]
```diff
@@ -584,11 +555,35 @@ private ImageSelector _getImageSelector(
...
 				}
-				finally {
-					StreamUtil.cleanUp(inputStream);
+			}
+		}
+
+		return null;
+	}
+
+	private InputStream _getInputStream(
```
Comment (Preston-Crary) [inline modules/apps/collaboration/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/exportimport/data/handler/BlogsEntryStagedModelDataHandler.java]: Let's rename this to _getImageSelectorInputStream

File: modules/apps/collaboration/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/verify/DLServiceVerifyProcess.java [hunk 2]
```diff
@@ -503,11 +503,13 @@ protected void doVerify() throws Exception {
 	protected String getMimeType(InputStream inputStream, String title) {
 		String mimeType = null;
 
-		try {
-			mimeType = MimeTypesUtil.getContentType(inputStream, title);
+		try (InputStream in = inputStream) {
```
Comment (Preston-Crary) [inline modules/apps/collaboration/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/verify/DLServiceVerifyProcess.java]: Let's rename 'in' to 'is'

File: modules/apps/collaboration/document-library/document-library-service/src/main/java/com/liferay/document/library/verify/DLServiceVerifyProcess.java [hunk 3]
```diff
@@ -525,11 +525,13 @@ protected void doVerify() throws Exception {
 	protected String getMimeType(InputStream inputStream, String title) {
 		String mimeType = null;
 
-		try {
-			mimeType = MimeTypesUtil.getContentType(inputStream, title);
+		try (InputStream in = inputStream) {
```
Comment (Preston-Crary) [inline modules/apps/collaboration/document-library/document-library-service/src/main/java/com/liferay/document/library/verify/DLServiceVerifyProcess.java]: in -> is

File: modules/apps/collaboration/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/pop/MessageListenerImpl.java [hunk 4]
```diff
@@ -249,9 +249,10 @@ public void deliver(String from, String recipient, Message message)
 				for (ObjectValuePair<String, InputStream> inputStreamOVP :
 						inputStreamOVPs) {
 
-					InputStream inputStream = inputStreamOVP.getValue();
-
-					StreamUtil.cleanUp(inputStream);
+					try (InputStream inputStream = inputStreamOVP.getValue()) {
+					}
+					catch (IOException ioe) {
```
Comment (Preston-Crary) [inline modules/apps/collaboration/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/pop/MessageListenerImpl.java]: Why does this one have the empty catch but the one in MBMessageStagedModelDataHandler.java does not?

File: modules/apps/collaboration/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/portlet/action/ImportPagesMVCActionCommand.java [hunk 5]
```diff
@@ -116,7 +115,8 @@ protected void importPages(
 		}
 		finally {
 			for (InputStream inputStream : inputStreams) {
-				StreamUtil.cleanUp(inputStream);
+				try (InputStream in = inputStream) {
```
Comment (Preston-Crary) [inline modules/apps/collaboration/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/portlet/action/ImportPagesMVCActionCommand.java]: For these cases, let's just call close() after doing a null check.

File: modules/apps/foundation/portal-store/portal-store-s3/src/main/java/com/liferay/portal/store/s3/S3FileCacheImpl.java [hunk 6]
```diff
@@ -107,33 +107,27 @@ public File getCacheFile(S3Object s3Object, String fileName)
...
 		File cacheFile = new File(cacheFileName);
 
-		InputStream inputStream = s3Object.getObjectContent();
+		try (InputStream inputStream = s3Object.getObjectContent()) {
+			if (cacheFile.exists() &&
+				(cacheFile.lastModified() >= lastModifiedDate.getTime())) {
 
-		if (cacheFile.exists() &&
-			(cacheFile.lastModified() >= lastModifiedDate.getTime())) {
+				try (InputStream autoCloseableInput = inputStream) {
```
Comment (Preston-Crary) [inline modules/apps/foundation/portal-store/portal-store-s3/src/main/java/com/liferay/portal/store/s3/S3FileCacheImpl.java]: This isn't needed, we don't need to do anything as it will be closed by the outer try-with-resources.

File: modules/apps/foundation/portal-store/portal-store-s3/src/main/java/com/liferay/portal/store/s3/S3Store.java [hunk 7]
```diff
@@ -372,17 +371,15 @@ public void updateFile(
 
 		File file = null;
 
-		try {
-			file = FileUtil.createTempFile(is);
+		try (InputStream in = is) {
```
Comment (Preston-Crary) [inline modules/apps/foundation/portal-store/portal-store-s3/src/main/java/com/liferay/portal/store/s3/S3Store.java]: This stream is closed inside of FileUtil, we can just remove the cleanUp call for this one.

File: modules/apps/knowledge-base/knowledge-base-api/src/main/java/com/liferay/knowledge/base/util/KnowledgeBaseUtil.java [hunk 8]
```diff
@@ -109,7 +106,7 @@ public static long getKBFolderId(
 	}
 
 	public static String getMimeType(byte[] bytes, String fileName) {
-		InputStream inputStream = new UnsyncByteArrayInputStream(bytes);
+		z
```
Comment (Preston-Crary) [inline modules/apps/knowledge-base/knowledge-base-api/src/main/java/com/liferay/knowledge/base/util/KnowledgeBaseUtil.java]: Compile error?

File: modules/apps/push-notifications/push-notifications-sender/push-notifications-sender-apple/src/main/java/com/liferay/push/notifications/sender/apple/internal/ApplePushNotificationsSender.java [hunk 9]
```diff
@@ -213,6 +200,18 @@ protected void deactivate() {
 		_apnsService = null;
 	}
 
+	private InputStream _getInputStream(String certificatePath) {
```
Comment (Preston-Crary) [inline modules/apps/push-notifications/push-notifications-sender/push-notifications-sender-apple/src/main/java/com/liferay/push/notifications/sender/apple/internal/ApplePushNotificationsSender.java]: Please rename this method to be more meaningful.

File: modules/apps/social-private-messaging/social-private-messaging-web/src/main/java/com/liferay/social/privatemessaging/web/internal/portlet/PrivateMessagingPortlet.java [hunk 10]
```diff
@@ -279,9 +278,8 @@ public void sendMessage(
 			for (ObjectValuePair<String, InputStream> inputStreamOVP :
 					inputStreamOVPs) {
 
-				InputStream inputStream = inputStreamOVP.getValue();
-
-				StreamUtil.cleanUp(inputStream);
+				try (InputStream inputStream = inputStreamOVP.getValue()) {
```
Comment (Preston-Crary) [inline modules/apps/social-private-messaging/social-private-messaging-web/src/main/java/com/liferay/social/privatemessaging/web/internal/portlet/PrivateMessagingPortlet.java]: Let's just call close() after null check.

File: modules/apps/sync/sync-service/src/main/java/com/liferay/sync/internal/servlet/SyncDownloadServlet.java [hunk 11]
```diff
@@ -588,6 +558,28 @@ protected void setUserLocalService(UserLocalService userLocalService) {
 		_userLocalService = userLocalService;
 	}
 
+	private InputStream _getInputStream(
```
Comment (Preston-Crary) [inline modules/apps/sync/sync-service/src/main/java/com/liferay/sync/internal/servlet/SyncDownloadServlet.java]: Rename this one too.

File: modules/apps/web-experience/export-import/export-import-service/src/main/java/com/liferay/exportimport/lar/ExportImportHelperImpl.java [hunk 12]
```diff
@@ -552,13 +551,14 @@ public ManifestSummary getManifestSummary(
...
-			fileEntry.getFileEntryId(), fileEntry.getVersion(), false);
+
 		ZipReader zipReader = null;
 
 		ManifestSummary manifestSummary = null;
 
-		try {
+		try (InputStream inputStream = _dlFileEntryLocalService.getFileAsStream(
+				fileEntry.getFileEntryId(), fileEntry.getVersion(), false))
+		{
```
Comment (Preston-Crary) [inline modules/apps/web-experience/export-import/export-import-service/src/main/java/com/liferay/exportimport/lar/ExportImportHelperImpl.java]: Please run the Source Formatter before sending a pull request.

File: modules/apps/web-experience/journal/journal-service/src/main/java/com/liferay/journal/exportimport/data/handler/JournalArticleStagedModelDataHandler.java [hunk 13]
```diff
@@ -1019,6 +999,33 @@ protected void setUserLocalService(UserLocalService userLocalService) {
 		_userLocalService = userLocalService;
 	}
 
+	private InputStream _getInputStream(
```
Comment (Preston-Crary) [inline modules/apps/web-experience/journal/journal-service/src/main/java/com/liferay/journal/exportimport/data/handler/JournalArticleStagedModelDataHandler.java]: Rename.

File: modules/apps/web-experience/layout/layout-set-prototype-impl/src/main/java/com/liferay/layout/set/prototype/internal/exportimport/data/handler/LayoutSetPrototypeStagedModelDataHandler.java [hunk 14]
```diff
@@ -312,22 +309,19 @@ protected void importLayouts(
...
+		try (InputStream inputStream =
+				portletDataContext.getZipEntryAsInputStream(
+					layoutSetPrototypeLARPath)) {
 
 			SitesUtil.importLayoutSetPrototype(
 				importedLayoutSetPrototype, inputStream, serviceContext);
 		}
-		finally {
-			StreamUtil.cleanUp(inputStream);
+		catch (IOException ioe) {
```
Comment (Preston-Crary) [inline modules/apps/web-experience/layout/layout-set-prototype-impl/src/main/java/com/liferay/layout/set/prototype/internal/exportimport/data/handler/LayoutSetPrototypeStagedModelDataHandler.java]: Why was this added?

File: portal-impl/src/com/liferay/portal/tools/deploy/ExtDeployer.java [hunk 15]
```diff
@@ -43,7 +44,10 @@ else if (arg.endsWith(".jar")) {
 			}
 		}
 
-		StreamUtil.cleanUp(new ExtDeployer(wars, jars));
+		try (ExtDeployer temp = new ExtDeployer(wars, jars)) {
+		}
+		catch (IOException ioe) {
```
Comment (Preston-Crary) [inline portal-impl/src/com/liferay/portal/tools/deploy/ExtDeployer.java]: Why does this one have an empty catch when the one in BaseDeployer.java logs?

File: portal-impl/src/com/liferay/portlet/documentlibrary/util/RawMetadataProcessorImpl.java [hunk 16]
```diff
@@ -158,8 +156,8 @@ public void saveMetadata(FileVersion fileVersion) throws PortalException {
 					fileVersion.getExtension(), fileVersion.getMimeType(),
 					inputStream);
 			}
-			finally {
-				StreamUtil.cleanUp(inputStream);
+			catch (IOException ioe) {
```
Comment (Preston-Crary) [inline portal-impl/src/com/liferay/portlet/documentlibrary/util/RawMetadataProcessorImpl.java]: We should not catch the exception.

**Comment by Preston-Crary** [comment]
Update the things I mentioned and check the rest of your code to see if the same changes are needed there as well.

Thanks

**Comment by lanceji92674957** [comment]
Doing it now.

File: portal-impl/src/com/liferay/portlet/documentlibrary/util/RawMetadataProcessorImpl.java (hunk 16 above)
Comment (lanceji92674957) [reply portal-impl/src/com/liferay/portlet/documentlibrary/util/RawMetadataProcessorImpl.java]: it's causing compile error since the exception is gonna be thrown by close() when auto closed. In these cases then, should I just add the IOException to method signature?

File: modules/apps/web-experience/layout/layout-set-prototype-impl/src/main/java/com/liferay/layout/set/prototype/internal/exportimport/data/handler/LayoutSetPrototypeStagedModelDataHandler.java (hunk 14 above)
Comment (lanceji92674957) [reply modules/apps/web-experience/layout/layout-set-prototype-impl/src/main/java/com/liferay/layout/set/prototype/internal/exportimport/data/handler/LayoutSetPrototypeStagedModelDataHandler.java]: try-with-resources will throw IOException, have to catch it to prevent compilation error
